assets/javascripts/job_templates.js

```diff
@@ -419,44 +419,46 @@ function submitTemplateEditor(preview) {
             reference: form.data('reference'),
         }
     }).done(function(data) {
-        result.text(data.preview ? 'Preview of the YAML:' : 'YAML saved!');
+        result.text(data.preview ? 'Preview of the changes:' : 'YAML saved!');
         if (!data.hasOwnProperty('preview')) {
             // Once a valid YAML template was saved we no longer offer the legacy editor
             $('#toggle-yaml-editor').hide();
             $('#media-add').hide();
         }
-        if (data.hasOwnProperty('template')) {
+        if (data.hasOwnProperty('changes')) {
             var preview = CodeMirror(result[0], {
-                mode: 'yaml',
+                mode: data.changes ? 'diff' : 'yaml',
                 lineNumbers: true,
                 lineWrapping: true,
                 readOnly: true,
-                value: data.template,
+                value: data.changes,
             });
         }
         else {
-            $('<pre/>').text(JSON.stringify(data, undefined, 2)).appendTo(result);
+            $('<b/>').text('No changes were made!').appendTo(result);
```

Comment:
I'd prefer using `strong` instead of `b`. And there should be a white-space before this message. Otherwise it follows directly after 'YAML saved!' with no spacing.